core/templates/dev/head/editor/ExplorationPreview.js

```diff
@@ -59,5 +61,31 @@ oppia.controller('ExplorationPreview', [
     $scope.$on('playerStateChange', function() {
       $scope.allParams = LearnerParamsService.getAllParams();
     });
+
+    explorationData.getData().then($scope._refreshPreviewTab = function() {
+      var newStateName = explorationInitStateNameService;
```

Comment:
Note that this is a misnomer; the service is not a name. Instead, you probably want something like:

```
    var initStateNameForPreview = explorationInitStateNameService.savedMemento;
    if (...) {
      initStateNameForPreview = ...;
    }
```

and then use initStateNameForPreview as the value for init_state_name when calling populateExploration().